tests/extract/test_extract_job.py

```diff
@@ -0,0 +1,69 @@
+from jx_base.expressions import NULL
+from jx_mysql.mysql import MySQL
+from jx_mysql.mysql_snowflake_extractor import MySqlSnowflakeExtractor
+from mo_files import File
+from mo_future import text
+from mo_sql import SQL
+from mo_testing.fuzzytestcase import assertAlmostEqual
+
+
+def test_make_repository(test_repository, extract_job_settings):
+    # TEST EXISTING FIXTURE MAKES AN OBJECT IN THE DATABASE
+    source = MySQL(extract_job_settings.source.database)
+    with source.transaction():
+        result = source.query(SQL("SELECT * from repository"))
+
+    # verify the repository object is the one we expect
+    assert result[0].id == test_repository.id
+    assert result[0].tc_root_url == test_repository.tc_root_url
+
+
+def test_make_failure_class(failure_class, extract_job_settings):
+    # TEST I CAN MAKE AN OBJECT IN THE DATABASE
+    source = MySQL(extract_job_settings.source.database)
+    with source.transaction():
+        result = source.query(SQL("SELECT * from failure_classification"))
+
+    # verify the repository object is the one we expect
+    assert result[0].name == "not classified"
+
+
+def test_make_job(complex_job, extract_job_settings):
+    source = MySQL(extract_job_settings.source.database)
+    with source.transaction():
+        result = source.query(SQL("SELECT count(1) as num from job_detail"))
+
+    assert result[0].num == 4
+
+
+# VERIFY SQL OVER DATABASE
+def test_extract_job_sql(extract_job_settings, transactional_db):
+    extractor = MySqlSnowflakeExtractor(extract_job_settings.source)
+    sql = extractor.get_sql(SQL("SELECT 0"))
+    assert "".join(sql.sql.split()) == "".join(EXTRACT_JOB_SQL.split())
```

Comment:
@camd Putting the SQL in its own file is prone to the auto-formatters reformatting it: Direct comparison no longer works; ignoring whitespace comparison works.